### ISSUE TITLE
How to use gcc-mcdc ?

### ISSUE THREAD

**Issue by Eeylx** [theHolgi/gcc-mcdc#1]
Hello, 
I really need this tool! I want to measure MC/DC with gcc, but I don't know how to use it.
please tell me, thank you very much!

**Comment by theHolgi** [comment]
Yes, I started working on MC/DC coverage instrumentation for GCC some while ago.
Unfortunately, I got stranded and didn't do anything for a long time.

I understood how the file format of .gcno and .gcda works, and how the compiler builds the GENERIC representation. The point where I'm stranded is how I would attach some information on a branch, about on which condition it belongs to, so that it survives GIMPLE translation and would later be available when we want to write the GCDA trace. 

Or, to explain it different:
To get from the arc coverage we already have to the MC/DC coverage, all we need is to record the structure of the expressions. So assume we have a:
(A && B && C)

This has 4 possible evaluations and therefore 4 branches that leave the expression:

(F && * && *)
(T && F && *)
(T && T && F)
(T && T && T)
So we need a record about this expression:
- Its structure
- 4 associations of possible evaluations with the arc ID that leads out of the expression.
